### DISCUSSION TITLE
Is SharpScript not possible to use commercially without buying the full product?

### DISCUSSION THREAD

**Discussion by simendsjo** [ServiceStack/Discuss#132] [Q&A]
I somehow assumed ServiceStack.Common was FOSS, but it looks like it's only licensed using the ServiceStack License.
This means I cannot embed ScriptSharp without paying $999/developer, or roughly $300,000 per year for my client.
Am I missing something, or is it really all or nothing here?

**Comment by mythz** [comment]
ServiceStack is offered under a dual [AGPL](https://github.com/ServiceStack/ServiceStack/blob/main/license.txt) and commercial license that is [Free for OSS Projects](https://servicestack.net/free).

For non-OSS projects a paid commercial license is only required to unlock the [free quotas](https://servicestack.net/download#free-quotas) in some server libraries. ServiceStack.Common (inc. #Script) and all its dependencies (e.g. ServiceStack.Text) do not have any have any free-quota/technical restrictions so it can be used unrestricted without a paid license, i.e. it can be used for Free.

> P.S. maximum price for ServiceStack for organizations with <500M revenue is $24,999 for an unlimited developers SITE License, that is further discounted 30% if renewed
